### ISSUE TITLE
[Feature Idea] Shopping List

### ISSUE THREAD

**Issue by lightlike** [nextcloud/cookbook#460]
**Is your feature request related to a problem? Please describe.**
Nope. No Problem.

**Context**
We are currently using the Deck-App as a simple shopping list.

**Describe the solution you'd like**
Could it be possible to add the functionality to add a recipes ingredients to this shopping list?
This would be nice if it would work with together with a manual shopping list (merge or certain tagged items).
Simple on button move card in Deck-App

**Idea**
Maybe all cookbook-items could be tagged in some way so cookbook knows what it can change and just move it on the board or create it when the recipe is added to the shopping list.

**Detailed info of the shopping list (for further clarification but pretty simple)**
- two columns: "to buy" and "bought"
- to buy has all the items for the current shopping trip
- when bought we move the card to the bought column to not buy it twice but so we do not have to add it back every time we need to buy it

**Comment by christianlupus** [comment]
This is similar to #127 in the sense that inter-app coordination would be required. I am not (yet) sure/convinced how this can be achieved in an easy way as the versions of the apps would be dependent on each other. In general, I see the problem that integration with the other apps are a wish.

**Comment by chriswells0** [comment]
First of all, thanks for the app: it really is fantastic!

Even without integrating into another app, it would be great if Cookbook could accumulate a list that can be sorted, copied/saved/printed, and then cleared/reset. For example, add the items and display them on a "Shopping List" page with checkboxes beside them. As long as the list is stored on the server side, they could simply open the list directly on their phone. Perhaps save the list as a markdown file in the recipes folder, which would allow checkboxes to be used inside the Nextcloud mobile app. Alternatively, the user could print the list to PDF and copy it to their phone.

**Comment by fds2610** [comment]
@chriswells0 idea is perfectly enough - no need to interface with other apps. functional requirements in this prio-order:
1. add recipies-ingredients to an internal shopping-list (markdown-file)
2. show shopping-list, clear list
3. manually delete items from shopping-list, if you have those items at home already
4. add checkbox to tick while in the shop to know what is already bought and what to look for
5. order shopping-list items automatically by type (e.g. vegetables, cooling, meat, cans, liquids, frozen, etc.)
6. combine same items by adding amounts (300ml milk + 250ml milk => 550ml milk)
7. when adding a recipe to the shopping-list, add a day-of-week keyword + add a week-view to see, when to cook what
all theese functions can be seen in Thermomix' cookidoo.de Website, if you have a login ;-)
+ sync shopping-list with android/iOS apps and provide similar functionality of editing lists

**Comment by Bugsbane** [comment]
While every item listed above would be great, just items 1,2 and 4 would make a perfectly workable initial version (items you have at home already could just be ticked off).

While I'd seriously love to see 6 happen, it could take quite a bit of processing to automatically figure out combinations of Tbs tbs "pinch of" grams gms g etc.

An easier first step might be to just let the user choose items to combine, take the description from the first one and ask the user the total amount for the combined ingredient item. Yes, they'd have to do a little basic maths themselves. I'd trust my own estimation of how much "a pinch of cinnamon" plus "5g of cinnamon" plus "1/2 tsp of cinnamon" is more than an algo. It's not hard and I'd rather have a clean list implemented than have this feature put off forever while the devs try to figure out natural language processing for every ingredient amount possible.

Maybe try creating the simple version first and then add basic automatic combination to that, which can be refined over time? Eventually the shopping list could ask if specific items should be combined where it's able to recognize them, and then the user could do the rest manually.

**Comment by MarcelRobitaille** [comment]
I just opened a duplicate issue without realizing. Sorry for the noise.

I outlined my ideas of how it would work there, which is still available. I can work on this issue once we figure out how it should work.

**Comment by rafny** [comment]
Responding to questions in #127 
@rafny, you want to add the appointment to an existing calendar or have the app to create a dedicated calendar for this issue? Accessing an existing one might be simpler...
- A dedicated calendar may be best since this can be shared among users/family members without having to share their personal calendars. Maybe on app installation, it creates the calendar that is managed directly.

The meal types (breakfast, lunch, ...) are pre-defined or user-configurable? Every day in the same time slot (maybe different on weekends)?
- I don't think the meal type has to be predefined. I can have a appetizer for lunch, brunch, or dinner. Instead, upon adding the meal there is a selection option for the type.

By task list you mean a task (as a calendar task) or a list in a notes file?
- Calendar
 
If calendar tasks: One would have to either have another calendar to register all the shopping tasks to or use the same calendar as for the meal planner itself.
- Yes, a calendar to see the meals and a calendar based on a Task List that is updated based on the recipe ingredients

I'm currently exploring https://grocy.info/ as an alternative and much of the functionality is there. Honestly, if that can be an app it would probably work great and just connect their task lists and calendars with NC.

**Comment by bpmartin20** [comment]
I agree with bugsbane regarding fds2610's list: 1,2 and 4 are the most important.

I would also add being able to manually add to the list (probably implied, but I wanted to mention it clearly) as a high priority.  

As a lesser priority, being able to group ingredients by store location, so that all the dairy items appear together on the list, and all the produce items appear elsewhere on the list. Just a simple location column that we can sort is fine, no fancy headers or anything.

I really like what I see of this app so far.  Good job, and please keep it up.

**Comment by Mousketeer** [comment]
This would be a game-changer for our family weekly grocery shopping trip, and I'm sure the same applies to a significant percentage of households.

May I suggest the output application would be configurable/pluggable for different Nextcloud apps?

After a few attempts we have settled on [Carnet](https://apps.nextcloud.com/apps/carnet) -hosted in Nextcloud of course- as a shopping-list tool for the user experience. Ticked boxes move out of the way, we can order/group according to store/stores layout and route. The sync isn't quasi-realtime so multiple concurrent shoppers need manual coordination.
For us this works far better than Cards, Joplin and plain notes.

**Comment by lordafzal** [comment]
I'm extremely excited about this feature. https://github.com/mhaendler already has a branch with a bunch of commits is that ready to be tested? Is there anything blocking that from being merged. I can try to help out if necessary. Should we work from his fork or just start again with a new fork is this close to going in but I don't understand the nextcloud PR cadences yet?
